From nobody Mon Sep 17 00:00:00 2001
Subject: [PATCH] Don't auto-reap traced children
From: Oleg Nesterov <oleg@tv-sign.ru>
Date: 1131632538 +0300

If a task is being traced we never auto-reap it even if it might look
like its parent doesn't care. The tracer obviously _does_ care.

Signed-off-by: Linus Torvalds <torvalds@osdl.org>
Signed-off-by: Chris Wright <chrisw@osdl.org>
---

 kernel/signal.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.14.y/kernel/signal.c
===================================================================
--- linux-2.6.14.y.orig/kernel/signal.c
+++ linux-2.6.14.y/kernel/signal.c
@@ -1524,7 +1524,7 @@ void do_notify_parent(struct task_struct
 
 	psig = tsk->parent->sighand;
 	spin_lock_irqsave(&psig->siglock, flags);
-	if (sig == SIGCHLD &&
+	if (!tsk->ptrace && sig == SIGCHLD &&
 	    (psig->action[SIGCHLD-1].sa.sa_handler == SIG_IGN ||
 	     (psig->action[SIGCHLD-1].sa.sa_flags & SA_NOCLDWAIT))) {
 		/*
